lib/ctcs2_to_junit.pm

```diff
@@ -99,12 +99,20 @@ sub generateXML {
             name      => $test,
             status    => $case_status,
             time      => $test_results{$test}->{time});
-        $writer->startTag('system-err');
-        $writer->characters("");
-        $writer->endTag('system-err');
-        $writer->startTag('system-out');
-        $writer->characters("");
-        $writer->endTag('system-out');
+        if ($case_status eq 'failure' || $case_status eq 'skipped') {
+            (my $test_path = $test) =~ s/-/\//;
+            $test_path = '/opt/log/' . $test_path;
+            my $test_out_content = script_output("cat $test_path\n", 600);
+            $writer->startTag('system-out');
+            $writer->characters($test_out_content);
+            $writer->endTag('system-out');
+            if ($case_status eq 'failure') {
+                my $test_err_content = script_output("echo '====out.bad log====\n';\ncat $test_path" . ".out.bad;\necho '\n====full log====\n';\ncat  $test_path" . ".full", 600);
```

Comment:
Possibly a potential problem if the logs are very big, do you need print all failed log on webui since we have the compressed log uploaded.

Reply:
All log included pass log before compressed is only 116.7 kB. The fail and skip log should be only xkB / xx kB. No need to worry about that. And all fail log is needed.